name.abuchen.portfolio/src/name/abuchen/portfolio/online/SecuritySearchProvider.java

```diff
@@ -9,6 +9,11 @@
 
 public interface SecuritySearchProvider
 {
+    /**
+     * Returns the technical identifier of the security search provider.
+     */
+    String getId();
```

Comment:
I understand that you need to get a specific search provider to set the API key. But i would propose to do it with the class name (like it is done for the dividend feed). That is a compile time dependency that makes it easier to change the code later.

I know the quote feed does it that way. But... a) it is old code which I could consider to refactor at some time and b) the ID is needed to store the configured quote provider in the XML file as I do not want to have the implementation class name in the XML file (there are already some cases where I accidentally leaked the class name into the XML and it is pain in the back to change).

```
    public static <F extends DividendFeed> F getDividendFeed(Class<F> feedType)
```

Reply:
Good idea. I was wondering why the DividendFeed lookup was not the same as the quote feed.